src/test/groovy/net/vivin/gradle/versioning/TagTaskSpecification.groovy

```diff
@@ -166,27 +166,43 @@ class TagTaskSpecification extends Specification {
         annotated << [false, true]
     }
 
-    @Unroll
-    def 'tag is skipped if #tagAndPushName is executed (annotated: #annotated)'() {
+    @Unroll("#testName")
+    def 'created annotated tag has expected message'() {
         given:
-        testRepository.origin = origin
-
-        and:
+        System.getenv()
         testRepository
             .makeChanges()
-            .commitAndTag('0.0.1', annotated)
+            .commitAndTag("1.0.0", true)
             .makeChanges()
             .commit()
 
+        and:
+        if(tagMessage) {
+            new File(gradleRunner.projectDir, 'build.gradle') << "tag { tagMessage = $tagMessage }"
+        }
+
         when:
-        def buildResult = gradleRunner.withArguments('-P', 'release', 'tag', tagAndPushName).build()
+        def arguments = ['-P', 'release', 'tag']
+
+        arguments.addAll additionalArguments
+        gradleRunner.withArguments(arguments).build()
 
         then:
-        buildResult.output.contains ':tag SKIPPED'
-        testRepository.headTag == '0.0.2'
-        origin.repository.tags.containsKey('0.0.2')
+        testRepository.headTagMessage == expectedTagMessage
 
         where:
-        [tagAndPushName, annotated] << [['tagAndPush', 'tAP'], [false, true]].combinations()
+
+        caseName                                 | tagMessage                         | additionalArguments     || expectedTagMessage
+        "when using default tagMessage"          | null                               | []                      || "v1.0.1"
+        "when using custom tagMessage"           | "{ \"version: awesome\" }"         | []                      || "version: awesome"
+        "when using system property closure"     | "fromSystemProperty(\"message\")"  | ["-Dmessage=test sys"]  || "test sys"
+        "when using project property closure"    | "fromProjectProperty(\"message\")" | ["-Pmessage=test proj"] || "test proj"
+        "without tagMessage"                     | "null"                             | []                      || null
+        "with closure returning null"            | "{ null }"                         | []                      || null
+        "with closure returning empty string"    | "{ \"\" }"                         | []                      || null
+        "with closure returning blank string"    | "{ \"    \" }"                     | []                      || null
+
```

Comment:
added some tests here and fixed the ones to the changed implementation (empty message)
also test against whether the tag is annotated and has the right name